persistence/data_models/data_models_enums.go

```diff
@@ -74,6 +74,7 @@ const (
 	ImmediateTaskTypeWaitUntil             ImmediateTaskType = 1
 	ImmediateTaskTypeExecute               ImmediateTaskType = 2
 	ImmediateTaskTypeNewLocalQueueMessages ImmediateTaskType = 3
+	ImmediateTaskTypeVisibility            ImmediateTaskType = 4
```

Comment:
I just realized an issue. The current immediate task queue is not order guarantee. But these visiblity tasks require ordering(otherwise you may change a exe status to completed before running). 

We may end up create another queue -- ImmediateOrderedTask. You can do this refactor separately